### PR TITLE
update: enforce bumping engine version whenever appropriate

### PR DESCRIPTION
**Pull request by jasondellaluce**
**What type of PR is this?**

/kind cleanup

/kind documentation

**Any specific area of the project related to this PR?**

/area CI

/area build

**What this PR does / why we need it**:

This follows-up https://github.com/falcosecurity/libs/pull/1273 and follows the same logic, applying it to the Falco engine version.

Accordingly, the formula for updating the engine checksum has been changed for catching more information. Moreover, two CI checks have been added:
- A strict one (that causes a failure) checking both the checksum and the engine version for consistency
- A weak one, that attempts catching changes in the Falco engine implementation and sends a warning message

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by leogr — APPROVED** [review]

File: RELEASE.md [hunk 1]
```diff
@@ -217,7 +217,7 @@ This section provides more details around the versioning of the components that
...
...uced to the rules files syntax and loading logic, and/or when `FALCO_ENGINE_CHECKSUM` has changed. The checksum is computed by considering the available rules fields (see currently supported [Falco fields](https://falco.org/docs/reference/rules/supported-fields/)), the event types (see currently supported [Falco events](https://falco.org/docs/reference/rules/supported-events/)), and the supported driver schema version. A checksum indicates that something was not available in previous engine versions. See the [rules release guidelines](https://github.com/falcosecurity/rules/blob/main/RELEASE.md#versioning-a-ruleset) to understand how this affects the versioning of Falco rules. Breaking changes introduced in the Falco engine are not necessarily tied to the drivers or libs versions. The version number must be incremented every time and only when a single change or an atomic group of changes - which meet the criteria described above - is included in the `master` branch. Thus, a version bump can occur multiple times during the development and testing phases of a given release cycle. A given version bump must not group multiple changes that occurred sporadically during the release cycle.
```
[the remark this answers is not in the document]
Comment (Andreagit97) [reply RELEASE.md:220]: i would remove the last phrase IMO it is not so clear, the previous one already states that we need to bump the version every time a libs update meets the criteria

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -35,52 +35,83 @@ jobs:
...
-      
-      - name: Install build dependencies
-        run: sudo DEBIAN_FRONTEND=noninteractive apt install libjq-dev libelf-dev libyaml-cpp-dev cmake build-essential git -y
-
-      - name: Install build dependencies (non-minimal)
-        if: matrix.minimal != 'minimal'
-        run: sudo DEBIAN_FRONTEND=noninteractive apt install libssl-dev libyaml-dev libc-ares-dev libprotobuf-dev protobuf-compiler libgrpc++-dev protobuf-compiler-grpc rpm libcurl4-openssl-dev linux-headers-$(uname -r) clang llvm -y
-      
-      - name: Prepare project
+      - name: Check Engine checksum
```
Comment (Andreagit97) [inline .github/workflows/ci.yml:72]: maybe I'm missing something but why do we need this check and the following one? just the following one is not enough?

Comment (Andreagit97) [reply .github/workflows/ci.yml:72]: ah no i got it, it checked the one generated from the code with the one written by the developer!

File: RELEASE.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jasondellaluce) [reply RELEASE.md:220]: Not that it's not just related to libs bumps, but also to changes in the engine code itself. That's why I kept it generic. What do you think?

**Comment by jasondellaluce** [comment]
> CI Build / check-engine-version (pull_request) Failing after 6s

This is of course a false positive. The checksum is changed just because I changed how it's computed.

**Review by Andreagit97 — APPROVED** [review]
/approve

**Pull request merged**